### PR TITLE
Set environment variables of web server to all workers

### PR DESCRIPTION
**Pull request by jeet4320**
## WHAT

Use [aws elasticbeanstalk update-environment](https://docs.aws.amazon.com/cli/latest/reference/elasticbeanstalk/update-environment.html) api to update environment variables

1. Pull env variables from webserver
2. Push those env variables to each worker

## WHY
We don't have to copy all env variables to each EB env manually now

### CODE REVIEW

File: lib/setenv_eb.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    self.config.cleanup()
+    return ['']
+
+  def set_environment_variables(self, env_vars):
+    for env in self.config.procfile.env_list():
+      if 'web' not in env:
+        os.chdir(env) # eb setenv can be run in directory which has run eb init and has .elasticbeanstalk folder in it
+        self.logger.info('#### Using environment: ' + env + ' ####')
+        self.logger.info('       Updating environment variables. It will take few minutes...')
+        run_single_command(' '.join(['eb', 'use', env, '--profile', DEFAULT_PROFILE])) # this updates 'branch-defaults' in .elasticbeanstalk/config.yml
```
Comment (whatisdot) [inline lib/setenv_eb.py]: Let's discover what API call the `ebcli` uses and use that instead. That way we don't need to worry about directories.

File: lib/setenv_eb.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+      self.set_environment_variables(env_vars)
+
+    except Exception as e:
+      self.config.cleanup()
+      raise e
+    self.config.cleanup()
+    return ['']
+
+  def set_environment_variables(self, env_vars):
```
Comment (whatisdot) [inline lib/setenv_eb.py]: We should also do this when we deploy.

File: lib/setenv_eb.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      args.environment = prompt_answer('Enter aws environment name: ')
+    return [args, None]
+  def command(self):
+    env_vars = self.get_environment_variables('web')
+    option_settings = []
+    for key in env_vars:
+      option_setting = {}
+      option_setting['OptionName'] = str(key)
+      option_setting['Namespace'] = 'aws:elasticbeanstalk:application:environment'
+      option_setting['Value'] = str(env_vars[key])
```
Comment (whatisdot) [inline lib/setenv_eb.py]: You can also format it like this:
```python
for key in env_vars:
  option_settings.append({
    'OptionName': str(key),
    'Namespace': 'aws:elasticbeanstalk:application:environment',
    'Value': str(env_vars[key])
  })
```

File: git_setenv_eb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+SCRIPT_DIR="$(dirname ${BASH_SOURCE[0]})"
```
Comment (nickstanish) [inline git_setenv_eb]: Why doesn't this have a bash shebang at the top?

File: lib/setenv_eb.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    parser.add_argument('--force-authenticate', action='store_true')
+    args = parser.parse_args()
+    if not args.environment:
+      args.environment = prompt_answer('Enter aws environment name: ')
+    return [args, None]
+  def command(self):
+    env_vars = self.get_environment_variables('web')
+    option_settings = []
+    for key in env_vars:
+      option_settings.append({
```
Comment (nickstanish) [inline lib/setenv_eb.py]: seems like a good opportunity to use map https://docs.python.org/3/library/functions.html#map

File: lib/setenv_eb.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    self.set_environment_variables(option_settings)
+    return ['']
+
+  def set_environment_variables(self, option_settings):
+    self.procfile = Procfile(self.project_name(), self.args.environment)
+    for env in self.procfile.env_list():
+      if 'web' not in env:
+        self.logger.info('Updating environment variables for ' + env)
+        EB.update_environment(self.project_name(), env, option_settings)
+    self.logger.info('#### Refer AWS web UI to monitor each environment update status ####')
```
Comment (nickstanish) [inline lib/setenv_eb.py]: nice bonus would be a url to the aws environment

File: lib/setenv_eb.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    env_vars = self.get_environment_variables('web')
+    option_settings = []
+    for key in env_vars:
+      option_settings.append({
+        'OptionName': str(key),
+        'Namespace': 'aws:elasticbeanstalk:application:environment',
+        'Value': str(env_vars[key])
+      })
+    self.set_environment_variables(option_settings)
+    return ['']
```
Comment (nickstanish) [inline lib/setenv_eb.py]: this seems like a strange return

File: git_setenv_eb (hunk 4 above)
Comment (jeet4320) [reply git_setenv_eb]: this file is created by `git_create_script` script and its same for all other files

**Pull request merged**